content/master/concepts/compositions.md

```diff
@@ -0,0 +1,993 @@
+{{</hint >}}
+
+For example, to check that a resource's 
+{{<hover label="NonEmpty" line="12">}}status.atProvider.state{{</hover>}}
+field isn't empty. 
+<!-- vale Google.WordList = YES -->
+
+```yaml {label="NonEmpty",copy-lines="none"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+# Removed for Brevity
+spec:
+  resources:
+  # Removed for Brevity
+    - name: my-resource
+      base:
+        # Removed for brevity
+      readinessChecks:
+        - type: NonEmpty
+          fieldPath: status.atProvider.state
+```
+
+{{<hint "tip" >}}
+Checking {{<hover label="NonEmpty" line="11">}}NonEmpty{{</hover>}} doesn't
+require a `match` field.
+{{< /hint >}} 
+
+#### Always consider a resource ready
+{{<hover label="none" line="11">}}None{{</hover>}} considers the 
+composed resource to be ready as soon as it's created. Crossplane doesn't wait
+for any other conditions before declaring the resource ready. 
+
+For example, consider 
+{{<hover label="none" line="7">}}my-resource{{</hover>}}
+ready as soon as it's created.
+
+
+```yaml {label="none",copy-lines="none"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+# Removed for Brevity
+spec:
+  resources:
+  # Removed for Brevity
+    - name: my-resource
+      base:
+        # Removed for brevity
+      readinessChecks:
+        - type: None
+```
+
+## Verify a Composition
+
+View all available Compositions with `kubectl get composition`.
+
+```shell {copy-lines="1"}
+kubectl get composition
+NAME                                       XR-KIND        XR-APIVERSION                         AGE
+xapps.aws.platformref.upbound.io           XApp           aws.platformref.upbound.io/v1alpha1   123m
+xclusters.aws.platformref.upbound.io       XCluster       aws.platformref.upbound.io/v1alpha1   123m
+xeks.aws.platformref.upbound.io            XEKS           aws.platformref.upbound.io/v1alpha1   123m
+xnetworks.aws.platformref.upbound.io       XNetwork       aws.platformref.upbound.io/v1alpha1   123m
+xservices.aws.platformref.upbound.io       XServices      aws.platformref.upbound.io/v1alpha1   123m
+xsqlinstances.aws.platformref.upbound.io   XSQLInstance   aws.platformref.upbound.io/v1alpha1   123m
+```
+
+The `XR-KIND` lists the Composite Resource `kind` that's allowed to use the
+Composition template.  
+The `XR-APIVERSION` lists the Composite Resource API versions allowed to use the
+Composition template. 
+
+{{<hint "note" >}}
+The output of `kubectl get composition` is different than `kubectl get
+composite`. 
+
+`kubectl get composition` lists all available Compositions.
+
+`kubectl get composite` lists all created Composite Resources and their related
+Composition. 
+{{< /hint >}}
+
+## Composition validation
+
+When creating a Composition Crossplane automatically validates specific
+parameters in the Composition.
+
+* All resources either use a `name` or don't. Compositions can't use both named
+  and unnamed resources.
```

Comment:
This is another best practice - always use names. I regret allowing resources to _not_ be named per https://github.com/crossplane/crossplane/issues/4130.